### PR TITLE
enhance: optimize TermExpr IN with SIMD batch filter for numeric types

### PR DESCRIPTION
**Pull request by xiaofan-luan**
## Summary

- **SIMD batch filter** for numeric TermExpr IN: runtime-dispatched (AVX-512 / AVX2 / SSE2 / NEON), significantly faster than hash set for small-to-medium IN lists, auto-switches to hash for large IN lists based on `simdLaneCount<T>() * 8` threshold
- **Go rewriter optimizations**: removed `shouldUseInExpr` threshold-based IN splitting (all IN now kept for SIMD), single-value IN \u2192 equality, OR-equals always merge to IN (\u2265 2 values), sort/dedup of IN values with NaN handling fix
- **String FilterChunk**: bypasses ValueType variant construction, supports heterogeneous `string_view` lookup via transparent hash (zero-copy for sealed/mmap segments)
- **SetElement\<bool\> specialization**: two-flag (`has_true_`/`has_false_`) implementation to avoid `ankerl::unordered_dense::hash<bool>` ASAN crash (wyhash reads 8 bytes from 1-byte bool). Unifies TermExpr and JsonContainsExpr bool paths through the same specialization
- **SetElement load factor**: set to 0.5 for better hash lookup performance

## Test plan

- Go unit tests: threshold removal, single-value folding, OR/AND combining, Timestamptz support, Geometry/Text blocking
- C++ unit tests: SimdBatchElement In() and FilterChunk for all numeric types, SetElement\<bool\> specialization (both `vector<bool>` and `GenericValue` constructors), string/string_view FilterChunk edge cases
- CI: ut-cpp, ut-go, e2e

issue: #48609

### CODE REVIEW

File: internal/parser/planparserv2/rewriter/entry.go [hunk 1]
```diff
@@ -105,6 +106,25 @@ func (v *visitor) visitUnaryExpr(expr *planpb.UnaryExpr) interface{} {
...
+	if col := expr.GetColumnInfo(); col != nil && col.GetDataType() == schemapb.DataType_Bool {
+		vals := expr.GetValues()
+		switch len(vals) {
+		case 0:
+			return newAlwaysFalseExpr()
+		case 1:
+			return newUnaryRangeExpr(col, planpb.OpType_Equal, vals[0])
+		default:
+			// After sortTermValues dedup, 2 bool values = {false, true} = all values
+			return newAlwaysTrueExpr()
```
[the remark this answers is not in the document]
Comment (zhengbuqian) [reply internal/parser/planparserv2/rewriter/entry.go]: +1 for codex.

for nullable bool field, `IN (true,false)` should be rewritten as `IS NOT NULL`. I have implemented comprehensive bool expr optimization in https://github.com/milvus-io/milvus/pull/48459, we can remove this optimization from this PR.
